include/swift/Basic/SymbolicLinks.h

```diff
@@ -0,0 +1,33 @@
+//===--- SymbolicLinks.h ----------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_SYMBOLICLINKS_H
+#define SWIFT_BASIC_SYMBOLICLINKS_H
+
+#include "swift/Basic/LLVM.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/VirtualFileSystem.h"
+#include <string>
+#include <optional>
+
+namespace swift {
+
+/// Tries to resolve symbolic links in a given path, but preserves
+/// substitute drives on Windows to avoid MAX_PATH issues.
+/// On failure, returns the original path.
+std::string resolveSymbolicLinks(llvm::StringRef InputPath,
+    llvm::vfs::FileSystem *FileSystem,
+    std::optional<llvm::sys::path::Style> Style = std::nullopt);
```

Comment:
Can you add a comment for `Style` here? And please use `llvm::Optional` for now. We're switching over, but today is not that day 😅